include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -90,6 +91,8 @@ struct MemoryPortsImpl {
   Value writeData;
   SmallVector<Value> addrPorts;
   Value writeEn;
+  bool seqReads = false;
```

Comment:
This boolean can be avoided by using `mlir::Value::operator bool()` to check whether `readEn` is invalid/undefined or not.